code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -170,12 +170,80 @@
 	return ..()
 
 // Secrete Resin
+
+/obj/effect/xeno/raising_structure
+	icon = 'icons/mob/actions.dmi'
+	icon_state = RESIN_WALL
+	var/buildable_typepath = /turf/closed/wall/resin/regenerating
+	var/timer_id
+	var/starting_time
+
+/obj/effect/xeno/raising_structure/attacked_by(obj/item/I, mob/user, def_zone)
+	if(!isxeno(user))
+		return
+	var/mob/living/carbon/xenomorph/the_antibuilder = user
+	if(!CHECK_BITFIELD(the_antibuilder.xeno_caste.caste_flags, CASTE_IS_BUILDER))
+		to_chat(the_antibuilder, span_notice("You are not capable of cancelling resin build-ups as a non-builder!"))
+	else
+		to_chat(the_antibuilder, span_notice("You cancel the building of the resin structure"))
+		log_admin("[user] has cancelled a xeno structure build-up as a xenomorph")
+		cancel_build()
+
+/obj/effect/xeno/raising_structure/proc/cancel_build()
+	SIGNAL_HANDLER
+	// Fix for clicking and cancelling by yourself due to the click signal
+	if(world.time - starting_time < 1 SECONDS)
+		return
+	deltimer(timer_id)
+	qdel(src)
+
+/obj/effect/xeno/raising_structure/proc/do_build()
```

Comment:
This is code duplicate from the action.

Two solutions: you either put this code in a global proc, or this effect does nothing and the logic is still on the action